src/MonitorProfiler/MainProfiler/MainProfiler.cpp

```diff
@@ -45,3 +63,31 @@ STDMETHODIMP MainProfiler::LoadAsNotficationOnly(BOOL *pbNotificationOnly)
 
     return S_OK;
 }
+
+HRESULT MainProfiler::InitializeEnvironment()
+{
+    m_pEnvironment.reset(new (nothrow) ProfilerEnvironment(m_pCorProfilerInfo));
+    IfNullRet(m_pEnvironment);
+
+    return S_OK;
+}
+
+HRESULT MainProfiler::InitializeLogging()
+{
+    // Create an aggregate logger to allow for multiple logging implementations
+    unique_ptr<AggregateLogger> pAggregateLogger(new (nothrow) AggregateLogger());
```

Comment:
I would like to say std::make_unique but I'm pretty sure we won't be able to cross compile it due to c++ version :(

Reply:
Correct